### PR TITLE
Document producer batch API

### PR DESCRIPTION
**Pull request by shargan**

### CODE REVIEW

File: docs/producer.rst [hunk 1]
```diff
@@ -103,3 +103,36 @@ containing fields:
...
+
+    # Batch will either be delivered or an unrecoverable error will occur.
+    # Cancelling this future will not cancel the send.
+    record = await fut
+
+While any number of batches may be created, only a single batch per partition
+is sent at a time. Additional calls to ``send_batch()`` against the same
+partition will wait for the inflight batch to be delivered before sending.
+
+Upon delivery, ``RecordMetadata.offset`` will match the batch's first message.
```
Comment (tvoinarovskyi) [inline docs/producer.rst]: Did you mean to write `record.offset`?

File: docs/examples.rst [hunk 2]
```diff
@@ -12,4 +12,4 @@ Examples
    Python 3.5 async syntax <examples/python35_examples>
    SSL usage <examples/ssl_consume_produce>
    Local state consumer <examples/local_state_consumer>
-
+   Batch producer <examples/batch_produce>
```
Comment (tvoinarovskyi) [inline docs/examples.rst]: Probably missed to add this file

File: docs/producer.rst (hunk 1 above)
Comment (shargan) [reply docs/producer.rst]: I did.

File: docs/examples.rst (hunk 2 above)
Comment (shargan) [reply docs/examples.rst]: Ah, thanks for catching that.

**Comment by tvoinarovskyi** [comment]
Cool!

**Pull request merged**